src/main/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpoint.java

```diff
@@ -79,6 +99,22 @@ public CaseSearchResult searchCases(
         return result;
     }
 
+    private List<String> getCaseTypeIds(List<String> caseTypeIds) {
+        if (isAConsolidationQuery(caseTypeIds)) {
+            return getCaseTypes();
+        }
+        return caseTypeIds;
+    }
+
+    private List<String> getCaseTypes() {
+        final List<String> roles = userService.getUserRolesJurisdictions();
+        return caseDefinitionRepository.getAllCaseTypesByJurisdictions(roles).get();
+    }
```

Comment:
Let's move this method into the existing class `CaseTypeService`. And have the controller delegate to it
The method could be called: `getCaseTypesFromIdamRoles`